lib/utils/note-title.js

```diff
@@ -10,14 +10,29 @@
 const noteTitleAndPreviewRegExp = /(\S[^\n]*)\s*(\S[^\n]*)?/;
 
 /**
+ * Matches the title and excerpt in a note's content skipping opening # from title
+ *
+ * Both the title and the excerpt are determined as
+ * content starting with something that isn't
+ * whitespace and leads up to a newline or line end
+ * @type {RegExp} matches a title and excerpt in note content skipping opening # from title
+ */
+const noteTitleAndPreviewMdRegExp = /[# ]*(\S[^\n]*)\s*(\S[^\n]*)?/;
```

Comment:
For reference, here's how we do it in the macOS app with a string replace:

https://github.com/Automattic/simplenote-macos/blob/develop/Simplenote/NSString%2BCondensing.m#L60

But I think the regex is fine too.

Reply:
Just a note that this works exactly the same as `noteTitleAndPreviewRegExp` because of the _kleene star_ at the beginning `[# ]*` - in other words, this RegExp will _always_ match when the other one matches

Reply:
Exactly. This one is just a little bit extended to catch `#` at the beginning. We can't join them to one because we don't want to omit `#` for not markdown notes.

Reply:
Thanks for clarifying @natanielcz - I had been _assuming_ that pulling it out on normal notes is also okay.